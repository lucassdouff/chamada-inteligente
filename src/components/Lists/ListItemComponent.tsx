import { View, Text, TouchableWithoutFeedback, Image } from "react-native";
import { ListDataModel } from "../../core/models/ListDataModel";

interface ListItemComponentProps {

    itemData: ListDataModel;
}


export default function ListItemComponent({itemData} : ListItemComponentProps){
    let color = '';
    if (itemData.info){
        if (itemData.info.percent!=null){
            color =itemData.info.percent>=75?' text-green-600':' text-red-600';
        } else {
           color = itemData.info.present?' text-green-600':' text-red-600';
        }
    }

    return (
        <View className='mt-2 border-spacing-y-2 py-3.5 flex-row justify-between border-b border-b-gray-300 w-[97%] items-center'>
            <View className='px-1 flex-row items-center gap-4'>
                {itemData.photo?<Image source={{uri:itemData.photo}}/>:<Image source={require('../../assets/perfil.png')}style={{width: 32, height: 32}}/>}
                
                <Text>{itemData.name}</Text>
            </View>
            {itemData.info?(itemData.info.present!=null?(
                <TouchableWithoutFeedback className='px-2 mr-1' onPress={itemData.info.action}>
                <Text className={'underline'+color}>
                    {itemData.info.present?'PRESENTE':'AUSENTE'}
                </Text>
            </TouchableWithoutFeedback>
            ):
            <Text className={'px-2 mr-1'+ color}>
                {itemData.info.percent}%
            </Text>):null
            }
        </View>
    )
}