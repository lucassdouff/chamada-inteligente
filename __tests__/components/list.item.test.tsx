import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import ListItemComponent from '../../src/components/Lists/ListItemComponent';

describe('ListItem component should render correctly and do some action', () => {
    
    it('renders correctly with sufficient presence', () => {
        render(<ListItemComponent itemData={{
            id: 1,
            id_course: 1,
            id_attendance: 1,
            enrollment: 123456,
            name: 'Student 1',
            info: {
                description: '75',
                action: () => {}
            },
        }} />);
        
        expect(screen.getByText("Student 1")).toBeTruthy();
        expect(screen.getByText("75%")).toBeTruthy();
    });

    it('renders correctly with insufficient presence', () => {
        render(<ListItemComponent itemData={{
            id: 1,
            id_course: 1,
            id_attendance: 1,
            enrollment: 123456,
            name: 'Student 1',
            info: {
                description: '25',
                action: () => {}
            },
        }} />);
        
        expect(screen.getByText("Student 1")).toBeTruthy();
        expect(screen.getByText("25%")).toBeTruthy();
    });

    it('performs action on click with sufficient presence', () => {

        const mockAction = jest.fn();

        render(<ListItemComponent itemData={{
            id: 1,
            id_course: 1,
            id_attendance: 1,
            enrollment: 123456,
            name: 'Student 1',
            info: {
                description: 'PRESENTE',
                action: mockAction
            },
        }} />);
        
        fireEvent.press(screen.getByText("PRESENTE"));
        expect(mockAction).toHaveBeenCalled();
    });

    it('performs action on click with insufficient presence', () => {

        const mockAction = jest.fn();

        render(<ListItemComponent itemData={{
            id: 1,
            id_course: 1,
            id_attendance: 1,
            enrollment: 123456,
            name: 'Student 1',
            info: {
                description: 'AUSENTE',
                action: mockAction
            },
        }} />);
        
        fireEvent.press(screen.getByText("AUSENTE"));
        expect(mockAction).toHaveBeenCalled();
    });
});
