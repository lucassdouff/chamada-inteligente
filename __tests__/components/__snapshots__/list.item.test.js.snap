// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should render correctly and do some action renders correctly 1`] = `
<View
  style={
    [
      {
        "marginTop": 8,
      },
      {
        "paddingBottom": 14,
        "paddingTop": 14,
      },
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
      {
        "borderBottomWidth": 1,
      },
      {
        "borderBottomColor": "#d1d5db",
      },
      {
        "width": "97%",
      },
      {
        "alignItems": "center",
      },
    ]
  }
  testID="list-item"
>
  <View
    style={
      [
        {
          "paddingLeft": 4,
          "paddingRight": 4,
        },
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "marginLeft": -16,
          "marginTop": -16,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        [
          {
            "height": 32,
            "width": 32,
          },
          [
            {
              "marginLeft": 16,
              "marginTop": 16,
            },
          ],
        ]
      }
    />
    <Text
      style={
        [
          {
            "marginLeft": 16,
            "marginTop": 16,
          },
        ]
      }
    >
      Student 1
    </Text>
  </View>
  <Text
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "textDecorationLine": "underline",
        },
        {
          "color": "#dc2626",
        },
      ]
    }
  >
    AUSENTE
  </Text>
</View>
`;
